hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.controller.js

```diff
@@ -362,32 +306,202 @@ define([
       });
     }
 
-    function contractChange (reasonId, date) {
-      $scope.$broadcast('hrjc-loader-show');
+    /**
+     * Converts a date string into a Date object (if string is not empty)
+     *
+     * @todo This should probably happen inside the service that returns the data #
+     *
+     * @param {string} dateString
+     * @param {Date/null}
+     */
+    function convertToDateObject (dateString) {
+      var dateObj = $filter('formatDate')(dateString, Date);
 
-      contractRevisionService.validateEffectiveDate({
+      return dateObj !== 'Unspecified' ? dateObj : dateString;
+    }
+
+    /**
+     * Fetch updated Health and Life Insurance Plan Types
+     *
+     * @return {Promise}
+     */
+    function fetchInsurancePlanTypes () {
+      return $q.all([
+        { name: 'hrjobcontract_health_health_plan_type', key: 'plan_type' },
+        { name: 'hrjobcontract_health_life_insurance_plan_type', key: 'plan_type_life_insurance' }
+      ].map(function (planTypeData) {
+        contractHealthService.getOptions(planTypeData.name, true)
+          .then(function (planTypes) {
+            $rootScope.options.health[planTypeData.key] = _.transform(planTypes, function (acc, type) {
+              acc[type.key] = type.value;
+            }, {});
+          });
+      }));
+    }
+
+    /**
+     * Moves the file to a trash list that will later be used to delete the files.
+     *
+     * @param {Number} index the index of the file in the section list.
+     * @param {String} entityName the section name where the file is stored.
+     */
+    function fileMoveToTrash (index, entityName) {
+      var entityFiles = $scope.files[entityName];
+      var entityFilesTrash = $scope.filesTrash[entityName];
+
+      entityFilesTrash.push(entityFiles[index]);
+      entityFiles.splice(index, 1);
+    }
+
+    /**
+     * Validates that all files are within the max file size limit.
+     */
+    function filesValidate () {
+      var entityName, fieldName, i, len, uploaderEntity, uploaderEntityField, uploaderEntityFieldQueue;
+      var fileMaxSize = $scope.fileMaxSize;
+      var uploader = $scope.uploader;
+      var isValid = true;
+
+      for (entityName in uploader) {
+        uploaderEntity = uploader[entityName];
+
+        for (fieldName in uploaderEntity) {
```

Comment:
Can we use lodash here _.each + indexes?

Reply:
@igorpavlov I refactored the whole function though I could not add tests for this one. I made sure the functionality remained the same though:

![anim](https://user-images.githubusercontent.com/1642119/36591314-fceb6db8-1867-11e8-82bd-b042a7e5bc79.gif)

And when I say "the same" it means that new files number start at 1 instead of the last number in the queue, and uploaded files can't be seen until you refresh the window (I'll create ticket for those and add tests to this method on that ticket)